.github/workflows/publish.yml

```diff
@@ -0,0 +1,27 @@
+name: Publish to NPM
+on:
+  release:
+    types: [published]
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Check out the repo'
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - name: 'Install Node.js'
+        uses: actions/setup-node@e33196f7422957bea03ed53f6fbb155025ffc7b8 # v3.7.0
+        with:
+          node-version-file: '.nvmrc'
+          registry-url: 'https://registry.npmjs.org'
+      - name: Install dependencies
+        run: npm ci
+        with:
+          node-version-file: '.nvmrc'
+      - name: Build the package
+        run: npm run build
```

Comment:
What does this do? 
We need to include "build" script in package.json.
e.g.:
"build": "tsc --project tsconfig.json"